lib/internal/Magento/Framework/App/View/Asset/Publisher.php

```diff
@@ -42,14 +42,33 @@ public function __construct(
      */
     public function publish(Asset\LocalInterface $asset)
     {
-        $dir = $this->filesystem->getDirectoryRead(DirectoryList::STATIC_VIEW);
-        if ($dir->isExist($asset->getPath())) {
+        if ($this->isFileEquals($asset)) {
             return true;
         }
 
         return $this->publishAsset($asset);
     }
 
+    public function isFileEquals(Asset\LocalInterface $asset)
+    {
+        $dir = $this->filesystem->getDirectoryRead(DirectoryList::STATIC_VIEW);
+        if ($dir->isExist($asset->getPath()) === false) {
+            return false;
+        }
+
+        $source = $asset->getSourceFile();
+        $destination = $dir->getAbsolutePath($asset->getPath());
+
+        $sourceSum = md5_file($source);
+        $destinationSum = md5_file($destination);
+
+        if ($sourceSum !== $destinationSum) {
+            return false;
+        }
+
+        return true;
+    }
```

Comment:
Ah, the "let's make present just a little bit worse to favour a future which might not come as soon as I think it would" argument. Fine with me.